src/react-components/object-list.js

```diff
@@ -0,0 +1,181 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import rootStyles from "../assets/stylesheets/ui-root.scss";
+import styles from "../assets/stylesheets/presence-list.scss";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBoxes } from "@fortawesome/free-solid-svg-icons/faBoxes";
+import { faBox } from "@fortawesome/free-solid-svg-icons/faBox";
+import { faVideo } from "@fortawesome/free-solid-svg-icons/faVideo";
+import { faImage } from "@fortawesome/free-solid-svg-icons/faImage";
+import { faNewspaper } from "@fortawesome/free-solid-svg-icons/faNewspaper";
+import { faQuestion } from "@fortawesome/free-solid-svg-icons/faQuestion";
+
+const SORT_ORDER_VIDEO = 0;
+const SORT_ORDER_IMAGE = 1;
+const SORT_ORDER_PDF = 2;
+const SORT_ORDER_MODEL = 3;
+const SORT_ORDER_UNIDENTIFIED = 4;
+function mediaSortOrder(el) {
+  if (el.components["media-video"]) return SORT_ORDER_VIDEO;
+  if (el.components["media-image"]) return SORT_ORDER_IMAGE;
+  if (el.components["media-pdf"]) return SORT_ORDER_PDF;
+  if (el.components["gltf-model-plus"]) return SORT_ORDER_MODEL;
+  return SORT_ORDER_UNIDENTIFIED;
+}
+
+function mediaSort(el1, el2) {
+  return mediaSortOrder(el1) - mediaSortOrder(el2);
+}
+
+const THUMBNAIL_TITLE = new Map([
+  [SORT_ORDER_VIDEO, "Video"],
+  [SORT_ORDER_IMAGE, "Image"],
+  [SORT_ORDER_PDF, "PDF"],
+  [SORT_ORDER_UNIDENTIFIED, "Unknown Media Type"],
+  [SORT_ORDER_MODEL, "Model"]
+]);
+
+const DISPLAY_IMAGE = new Map([
+  [SORT_ORDER_VIDEO, faVideo],
+  [SORT_ORDER_IMAGE, faImage],
+  [SORT_ORDER_PDF, faNewspaper],
+  [SORT_ORDER_UNIDENTIFIED, faQuestion],
+  [SORT_ORDER_MODEL, faBox]
+]);
+function getDisplayImage(el) {
+  return DISPLAY_IMAGE.get(mediaSortOrder(el));
+}
+
+function getDisplayString(el) {
+  const url = el.components["media-loader"].data.src;
+  const split = url.split("/");
+  const resourceName = split[split.length - 1].split("?")[0];
+  let httpIndex = -1;
+  for (let i = 0; i < split.length; i++) {
+    if (split[i].indexOf("http") !== -1) {
+      httpIndex = i;
+    }
+  }
+
+  let host = "";
+  let lessHost = "";
+  if (httpIndex !== -1 && split.length > httpIndex + 3) {
+    host = split[httpIndex + 2];
+    const hostSplit = host.split(".");
+    if (hostSplit.length > 1) {
+      lessHost = `${hostSplit[hostSplit.length - 2]}.${hostSplit[hostSplit.length - 1]}`;
+    }
+  }
+
+  return `${lessHost} ... ${resourceName.substr(0, 30)}`;
+}
```

Comment:
I think this entire function would be easier to understand/maintain if you converted it into a regex instead of all these string splits.